File: source/com/barrybecker4/puzzle/tantrix/model/HexTile.java
// Copyright by Barry G. Becker, 2012. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.puzzle.tantrix.model;

import java.util.List;

/**
 * Represents a single tantrix tile.
 * Immutable.
 * @author Barry Becker
 */
public class HexTile {

    /** number of sides on a hex tile. */
    public static final byte NUM_SIDES = 6;

    private byte tantrixNumber;
    private List<PathColor> edgeColors;
    private PathColor primaryColor;

    /**
     * Constructor.
     */
    public HexTile(byte tantrixNumber, PathColor primaryColor, List<PathColor> edgeColors) {
        assert edgeColors.size() == 6;
        this.tantrixNumber = tantrixNumber;
        this.primaryColor = primaryColor;
        this.edgeColors = edgeColors;
    }

    /** @return the number on the back of the tile */
    public byte getTantrixNumber() {
        return tantrixNumber;
    }

    public PathColor getEdgeColor(int index) {
        return edgeColors.get(index);
    }

    /** The primary path color on the back of the tile */
    public PathColor getPrimaryColor() {
        return primaryColor;
    }

    public String toString() {
        return "tileNum=" + tantrixNumber + " colors: " + edgeColors;
    }

}

File: scala-source/com/barrybecker4/puzzle/tantrix/model/HexTile.scala
// Copyright by Barry G. Becker, 2017. Licensed under MIT License: http://www.opensource.org/licenses/MIT
package com.barrybecker4.puzzle.tantrix.model

import com.barrybecker4.puzzle.tantrix.model.PathColor

object HexTile {
  /** number of sides on a hex tile. */
  val NUM_SIDES = 6
}

/** Represents a single tantrix tile. Immutable.
  * @param tantrixNumber the number on the back of the tile
  * @param primaryColor  The primary path color on the back of the tile
  * @param edgeColors outgoing path colors for each orientation index
  * @author Barry Becker
  */
case class HexTile(tantrixNumber: Byte, primaryColor: PathColor, edgeColors: PathColors) {
  override def toString: String = "tileNum=" + tantrixNumber + " colors: " + edgeColors
}
